File: packages/create-react-native-library/templates/java-view-mixed/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}ViewManager.java
package com.<%- project.package %>;

import android.graphics.Color;

import androidx.annotation.Nullable;

import com.facebook.react.module.annotations.ReactModule;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

@ReactModule(name = <%- project.name -%>ViewManager.NAME)
public class <%- project.name -%>ViewManager extends <%- project.name -%>ViewManagerSpec<<%- project.name -%>View> {

  public static final String NAME = "<%- project.name -%>View";

  @Override
  public String getName() {
    return NAME;
  }

  @Override
  public <%- project.name -%>View createViewInstance(ThemedReactContext context) {
    return new <%- project.name -%>View(context);
  }

  @Override
  @ReactProp(name = "color")
  public void setColor(<%- project.name -%>View view, @Nullable String color) {
    view.setBackgroundColor(Color.parseColor(color));
  }
}

File: packages/create-react-native-library/templates/kotlin-view-mixed/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}ViewManager.kt
package com.<%- project.package %>

import android.graphics.Color
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

@ReactModule(name = <%- project.name -%>ViewManager.NAME)
class <%- project.name -%>ViewManager :
  <%- project.name -%>ViewManagerSpec<<%- project.name -%>View>() {
  override fun getName(): String {
    return NAME
  }

  public override fun createViewInstance(context: ThemedReactContext): <%- project.name -%>View {
    return <%- project.name -%>View(context)
  }

  @ReactProp(name = "color")
  override fun setColor(view: <%- project.name -%>View?, color: String?) {
    view?.setBackgroundColor(Color.parseColor(color))
  }

  companion object {
    const val NAME = "<%- project.name -%>View"
  }
}
